src/Microsoft.TemplateEngine.Cli/Commands/InstantiateCommand.cs

```diff
@@ -5,42 +5,277 @@
+                    out bool languageOptionSpecified);
+
+                //if no candidates continue with next precedence
+                if (!candidates.Any())
+                {
+                    continue;
+                }
+                //if language option is not specified, we do not need to do reparsing for default language
+                if (languageOptionSpecified || string.IsNullOrWhiteSpace(environmentSettings.GetDefaultLanguage()))
+                {
+                    return candidates;
+                }
 
+                // try to reparse for default language
+                return ReparseForDefaultLanguage(
+                    args,
+                    environmentSettings,
+                    templatePackageManager,
+                    templateGroup,
+                    candidates);
+            }
+            return new HashSet<TemplateCommand>();
         }
 
-        protected override Task<NewCommandStatus> ExecuteAsync(InstantiateCommandArgs args, IEngineEnvironmentSettings environmentSettings, InvocationContext context) => throw new NotImplementedException();
+        protected async override Task<NewCommandStatus> ExecuteAsync(InstantiateCommandArgs instantiateArgs, IEngineEnvironmentSettings environmentSettings, InvocationContext context)
+        {
+            //TODO: revise this output
+            //help doesn't have argumentes defined at the moment though should
+            //help for dotnet new should have both arguments and commands present
+            if (string.IsNullOrWhiteSpace(instantiateArgs.ShortName) && instantiateArgs.HelpRequested)
+            {
+                context.HelpBuilder.Write(
+                    instantiateArgs.ParseResult.CommandResult.Command,
+                    StandardStreamWriter.Create(context.Console.Out),
+                    instantiateArgs.ParseResult);
+                return NewCommandStatus.Success;
+            }
+
+            using TemplatePackageManager templatePackageManager = new TemplatePackageManager(environmentSettings);
+            var hostSpecificDataLoader = new HostSpecificDataLoader(environmentSettings);
+            if (string.IsNullOrWhiteSpace(instantiateArgs.ShortName))
+            {
+                TemplateListCoordinator templateListCoordinator = new TemplateListCoordinator(
+                    environmentSettings,
+                    templatePackageManager,
+                    hostSpecificDataLoader,
+                    TelemetryLogger);
+
+                return await templateListCoordinator.DisplayCommandDescriptionAsync(instantiateArgs, default).ConfigureAwait(false);
+            }
+
+            var templates = await templatePackageManager.GetTemplatesAsync(context.GetCancellationToken()).ConfigureAwait(false);
+            var templateGroups = TemplateGroup.FromTemplateList(CliTemplateInfo.FromTemplateInfo(templates, hostSpecificDataLoader));
+
+            //TODO: decide what to do if there are more than 1 group.
```

Comment:
I believe this should be solved elsewhere, when TemplateGroups are created... it should be checked for conflict and in that case have `ShortName`=`GroundIndentifier`, ugly but whole code base doesn't need to care about this then(like --list)...

Reply:
`list` is not really impacted by it, it is possible to show template groups with the same name. But, in general I agree.
It also depends on how we would like to solve it:
- if we want to prefix conflicted short name with group identity: `<group identity>:<short-name>`
- or require it as additional parameter: `<short-name> --group <group identity>`